pkg/util/convert/convert.go

```diff
@@ -0,0 +1,87 @@
+package convert
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+
+	kudov1beta1 "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+)
+
+// String returns a pointer to the string value passed in.
+func String(v string) *string {
+	return &v
+}
+
+// StringValue returns the value of the string pointer passed in or
+// "" if the pointer is nil.
+func StringValue(v *string) string {
+	if v != nil {
+		return *v
+	}
+	return ""
+}
+
+// UnwrapParamValue unwraps a parameter value.
+// Depending on the parameter type, the value can represent a string or an object described in YAML.
+func UnwrapParamValue(v *string, t kudov1beta1.ParameterType) (r interface{}, err error) {
+	switch t {
+	case kudov1beta1.MapValueType:
+		r, err = ToYAMLMap(StringValue(v))
+	case kudov1beta1.ArrayValueType:
+		r, err = ToYAMLArray(StringValue(v))
+	case kudov1beta1.StringValueType:
+		fallthrough
+	default:
+		r = StringValue(v)
+	}
+
+	return
+}
+
+// WrapParamValue wraps a parameter value.
+func WrapParamValue(i interface{}, t kudov1beta1.ParameterType) (*string, error) {
+	switch t {
+	case kudov1beta1.MapValueType:
+		fallthrough
+	case kudov1beta1.ArrayValueType:
+		bytes, err := yaml.Marshal(i)
+		if err != nil {
+			return nil, err
+		}
+
+		result := string(bytes)
+		return &result, nil
+	case kudov1beta1.StringValueType:
+		fallthrough
+	default:
+		if i == nil {
+			return nil, nil
+		}
+
+		result := fmt.Sprintf("%v", i)
+		return &result, nil
+	}
+}
+
+// ToYAMLArray converts YAML input describing an array.
```

Comment:
```suggestion
// ToYAMLArray unmarshals stringified YAML into an array.
```

I assume you realize that `ToYAMLArray` and `ToYAMLMap` are identical functions (with the only exception being the function name).   Would it make sense to valuate the type?   would it make sense to have one function that both of these call?   would it make sense to collapse them?

Reply:
They aren't, `ToYAMLArray` unmarshals to `[]interface{}` while `ToYAMLMap` unmarshals to `interface{}`. This difference is subtle though and we could use `ToYAMLMap` for arrays as well, as the difference between the `array` and `map` parameter types will be mostly around validation -- which this PR doesn't introduce.

Reply:
@nfnt I still think the godoc suggested change makes more sense

Reply:
Thanks, I didn't see the code suggestion. Agree, much better!